### DISCUSSION TITLE
Question about BCs in RVE Simulation

### DISCUSSION THREAD

**Discussion by stonebeta** [idaholab/moose#27163] [Q&A Modules: Solid mechanics]
In the homogenization test input file, [small-tests](https://github.com/idaholab/moose/tree/next/modules/solid_mechanics/test/tests/lagrangian/cartesian/total/homogenization/small-tests/3d.i),
there are constraints applied to two additional points ('fix_xy' and 'fix_z') in addition to the fixed point ('fix_all'). I am uncertain whether these two points should be constrained when all `constraint_types` are strain type. Could someone please clarify why these two extra points are constrained?

**Comment by GiudGiud** [comment]
Hello

We typically add extra constratins on displacements to avoid homogeneous solutions on displacements. So displacement modes that are also solution of the equations but moving the geometry around. They come from the null space of the equations. 
Someone more expert on RVE will have to check whether this is the case here
Guillaume

**Reply by RossMech** [reply]
As of my understanding, fixing one point resolves in restriction of translational modes and the additional two points should be restricted to remove the possibility of the rotation.

**Reply by RossMech** [reply]
There is 6 possible DOFs (3 translations and 3 rotations). Therefore, to restrict all of them, 6 conditions are needed.

**Reply by stonebeta** [reply]
So the second point should be very close to the first point?

**Reply by GiudGiud** [reply]
I dont think that needs to be the case.
You just need to block extraneous rotations and translations, and this can be done a number a ways.

For example with blocking the displacement (with dirichlet BCs) of three points that are not aligned
